### PR TITLE
Julia's temporary Main branch

### PR DESCRIPTION
**Pull request by julia-script**
# 🧙 Description
Since everybody is out now, I created this branch to merge my non urgent changes so I don't get blocked

Once someone with time is back, they can review it and we can merge this into main

### CODE REVIEW

**Comment by sandoche** [comment]
Would be nice for the users to be able to switch to testnet (on staging only) for the mods or our team to test things with fake money!

@julia-script is it an easy change?

**Review by sandoche — APPROVED** [review]
I approve there are just a few little things to fix.
Impressive work. Well done!

**Review by Mrsirdev — DISMISSED** [review]
Loooks awesome!

Havent had the time to look into the details yet.

Could you please add the package version bumps and add info to the CHANGELOG 🙏 

Great Work WIZZ 🧙‍♀️

File: packages/evmos-wallet/src/internal/wallet/functionality/networkConfig.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const EVMOS_CHAIN = {
 export const EVMOS_NETWORK_FOR_BACKEND =
   process.env.NEXT_PUBLIC_EVMOS_NETWORK_FOR_BACKEND ?? "EVMOS";
 export const EVMOS_BACKEND =
-  process.env.NEXT_PUBLIC_EVMOS_BACKEND ?? "https://goapi-staging.evmos.org";
+  process.env.NEXT_PUBLIC_EVMOS_BACKEND ?? "https://goapi.evmos.org";
```
Comment (miliPaoletti) [inline packages/evmos-wallet/src/internal/wallet/functionality/networkConfig.ts:26]: why did you remove staging from here ?

File: packages/evmos-wallet/src/registry-actions/hooks/use-fee.ts [hunk 2]
```diff
@@ -48,8 +48,9 @@ export const useFee = ({
             ...token,
             amount: 1n,
           },
-        }),
+        })
       );
+      console.log(err);
```
Comment (miliPaoletti) [inline packages/evmos-wallet/src/registry-actions/hooks/use-fee.ts:53]: is this clg necessary?

File: packages/evmos-wallet/src/wallet/wagmi/keplrConnector.ts [hunk 3]
```diff
@@ -178,18 +174,19 @@ export class KeplrConnector extends Connector<Keplr, {}> {
...
 
     const account = await this.getAccount();
-    let chainId = config?.chainId;
-    if (!chainId || !(chainId in COSMOS_ID_MAP)) {
-      chainId = this.chains.find(({ id }) => id in COSMOS_ID_MAP)?.id;
-      assertIf(chainId, "UNSUPPORTED_NETWORK");
-    }
+    const chainId = config?.chainId ?? evmosInfo.id;
+
+    // if (!chainId ) {
```
Comment (miliPaoletti) [inline packages/evmos-wallet/src/wallet/wagmi/keplrConnector.ts:179]: Please remove it if it is not necessary

**Review by miliPaoletti — APPROVED** [review]
Awesome job Julia!!!! 🚀 🎸

File: apps/assets/pages/register-token/index.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    }) => {
+      if (!address) {
+        throw new Error("No address");
+      }
+      const msg = new RegisterERC20Proposal({
+        title,
+        description,
+        erc20addresses: [erc20address],
+      });
+      // const msg = new RegisterCoinProposal({
```
[the remark this answers is not in the document]
Comment (julia-script) [reply apps/assets/pages/register-token/index.tsx:40]: good catch

File: packages/evmos-wallet/src/internal/wallet/functionality/networkConfig.ts [hunk 5]
```diff
@@ -2,14 +2,14 @@
 // SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/apps/blob/main/LICENSE)
 
 export const EVMOS_RPC_URL =
-  process.env.NEXT_PUBLIC_EVMOS_RPC_URL ?? "https://evmos.lava.build/";
+  process.env.NEXT_PUBLIC_EVMOS_RPC_URL || "https://evmos.lava.build/";
```
[the remark this answers is not in the document]
Comment (julia-script) [reply packages/evmos-wallet/src/internal/wallet/functionality/networkConfig.ts:5]: because of this..
image

if the value is an empty string (if I'm not mistaken, that's the case if you have the env var with no value in the env file), it will not fallback to the default value

File: packages/evmos-wallet/src/internal/wallet/functionality/networkConfig.ts (hunk 1 above)
Comment (julia-script) [reply packages/evmos-wallet/src/internal/wallet/functionality/networkConfig.ts:26]: oh, cause I thought staging was unstable, but both were unstable, I'll rever this, good catch :)

File: packages/evmos-wallet/src/registry-actions/hooks/use-fee.ts (hunk 2 above)
Comment (julia-script) [reply packages/evmos-wallet/src/registry-actions/hooks/use-fee.ts:53]: omg we need a lint rule for this haha

**Comment by julia-script** [comment]
> Would be nice for the users to be able to switch to testnet (on staging only) for the mods or our team to test things with fake money!
@julia-script is it an easy change?
@sandoche yes, if we just need to enable this ver
`NEXT_PUBLIC_ENABLE_TESTNET`
then this would show up
image

of course, local wouldn't work so we may want to hide it in staging

File: packages/evmos-wallet/src/registry-actions/hooks/use-fee.ts (hunk 2 above)
Comment (miliPaoletti) [reply packages/evmos-wallet/src/registry-actions/hooks/use-fee.ts:53]: hahahahh
